src/stats.rs

```diff
@@ -37,32 +37,26 @@ impl ProjectStats {
     }
 
     /// Checks whether this project exceeds its budgets.
-    ///
-    /// This will also update internal state when checking.
     pub fn exceeds_budget(&mut self) -> bool {
-        self.update_aggregated_state(self.config.truncated_now())
+        self.check_budget(self.config.truncated_now())
     }
 
     /// Records spent budget.
     ///
     /// This will also update internal state when checking.
-    pub fn record_budget_spend(&mut self, spent_budget: f64) -> bool {
+    pub fn record_spending(&mut self, spent: f64) -> bool {
         let now = self.config.truncated_now();
 
-        if let Some(latest) = self.budget_buckets.front_mut() {
-            if latest.0 >= now {
-                latest.1 += spent_budget;
-            } else {
-                if self.budget_buckets.len() >= self.config.num_buckets {
-                    self.budget_buckets.pop_back();
-                }
-                self.budget_buckets.push_front((now, spent_budget));
-            }
-        } else {
-            self.budget_buckets.push_front((now, spent_budget));
+        match self.budget_buckets.front_mut() {
+            Some(latest) if latest.0 >= now => latest.1 += spent,
+            _ => self.budget_buckets.push_front((now, spent)),
```

Comment:
please make sure to adjust the `with_capacity` call then as well, as this would double that capacity. We have a hard limit to how many buckets we are using, so we should not allow for any slack.

Reply:
Good catch, yeah.